consensus/reactor.go

```diff
@@ -658,6 +711,9 @@ func (conR *Reactor) updateRoundStateRoutine() {
 		conR.mtx.Lock()
 		conR.rs = rs
 		conR.mtx.Unlock()
+		
+		// Check if we should disable the old propagation routine
+		conR.checkAndDisableOldPropagationRoutine()
```

Comment:
the app version can only change after committing a block, so that's where the check must occur.

we definitely cannot take the mutex of the consensus reactor every 100ms for this, that is absolutely out of the question